src/main/java/com/govac/institutii/kafka/Producer.java

```diff
@@ -0,0 +1,57 @@
+package com.govac.institutii.kafka;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Producer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Producer.class);
+
+    private final KafkaProducer<byte[], byte[]> producer;
+
+    public Producer(Properties kafkaProperties) throws IOException {
+        producer = new KafkaProducer<byte[], byte[]>(kafkaProperties);
+    }
+
+    public void send(Message message) {
+        producer.send(new ProducerRecord<byte[], byte[]>(message.getTopic(), message.getKey(), message.getValue()),
+                callBack(message));
```

Comment:
problema e insa de control: am vrea sa facem rollback la intreaga tranzactie de adaugare a mesajului institutiei in db si sa returnam un mesaj de eroare in api, ideal. pentru asta ne trebuie confirmare sincrona. s-ar putea face prin state, banuiesc, dar nu e ideal